src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -77,39 +145,247 @@ public PolicyEntity updatePolicy(@PathParam("domainid") Long domainid, @PathPara
+                if (result.isEmpty()) {
+                    throw new Exception("Policy or Domain Entity does not exists...");
+                    //TODO - Add 404 error response - Sudhir
+                }
+                return result.get(0);
+            }
+        });
+    }
+
+    /**
+     * Creates the Domain for a given ApplianceManagerConnector
+     *
+     * @return Domain
+     */
+    @POST
+    public String createDomain(DomainEntity entity) {
+
+        logger.info("Creating Domain Entity...:" + entity.getName());
+
+        return this.txControl.required(new Callable<DomainEntity>() {
+
+            @Override
+            public DomainEntity call() throws Exception {
+
+                CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+                CriteriaQuery<DomainEntity> query = criteriaBuilder.createQuery(DomainEntity.class);
+                Root<DomainEntity> r = query.from(DomainEntity.class);
+                query.select(r).where(criteriaBuilder.and(criteriaBuilder.equal(r.get("name"), entity.getName())));
+
+                List<DomainEntity> result = em.createQuery(query).getResultList();
+                if (!result.isEmpty()) {
+                    throw new Exception("Domain name already exists...");
+                    //TODO - to add RETURN 400 error:Sudhir
+                }
+                em.persist(entity);
+                return entity;
+            }
+        }).getId();
+    }
+
+    /**
+     * Updates the Domain for a given domain Id
+     *
+     * @return - updated Domain
+     */
+    @Path("/{domainid}")
+    @PUT
+    public DomainEntity updateDomain(@PathParam("domainid") Long domainid, DomainEntity entity) {
+
+        logger.info("Updating Domain Entity ID...:" + domainid);
+
+        return this.txControl.required(new Callable<DomainEntity>() {
+
+            @Override
+            public DomainEntity call() throws Exception {
+
+                DomainEntity result = em.find(DomainEntity.class, domainid);
+                if (result == null) {
+                    throw new Exception("Domain Entity does not exists...");
+                    //TODO - to add RETURN 404 error:Sudhir
+                }
+                result.setName(entity.getName());
+                em.persist(result);
+                return result;
+            }
+        });
+    }
+
+    /**
+     * Deletes the Domain for a given domain Id
+     *
+     * @return - deleted Domain
+     */
+    @Path("/{domainid}")
+    @DELETE
+    public void deleteDomain(@PathParam("domainid") Long domainid) {
+
+        logger.info("Deleting Domain Entity ID...:" + domainid);
+
+        this.txControl.required(new Callable<Void>() {
+
+            @Override
+            public Void call() throws Exception {
+
+                DomainEntity result = em.find(DomainEntity.class, domainid);
+                if (result == null) {
+                    return null;
```

Comment:
return null; -> return; no need to return null for a void method. Please also check other places.

Reply:
txControl.required(new Callable<Void>() takes Java Void as an arg parameter. This expects the null to be returned. otherwise the compiler throws syntax error. This is also how it was done originally by Paremus in Device code.